backend/src/db/repositories/OAuth2Repository.ts

```diff
@@ -0,0 +1,213 @@
+import DB from '../DB';
+import { OAuth2ClientRaw } from '../types/OAuth2';
+import { ResultSetHeader } from 'mysql2';
+
+export default class OAuth2Repository {
+  private db: DB;
+
+  // Define a base SQL query that both methods share.
+  private clientBaseQuery = `
+    select 
+      oauth_clients.*,
+      oauth_consents.scope as scopes,
+      oauth_consents.last_revoked_ts as last_revoked_ts,
+      (select count(oauth_consents.user_id) from oauth_consents where client_id = oauth_clients.client_id) as installations_count 
```

Comment:
this could lead to performance issue if lots of apps are rendered at the same time

Reply:
fixed by caching locally